README.md

```diff
@@ -9,57 +9,314 @@
+- [Example Code (w/ Claims)](#example-code-with-claim-based-settlement)
+- [Example Code (w/ Payments)](#example-code-with-unconditional-payment-based-settlement)
+- [Extended Plugin Backend API](#backend-api-with-extensions-for-payment-channels)
+- [Payment Channel Backend API](#payment-channel-backend-api)
+- [Plugin Context API](#plugin-context-api)
+
+## Example Code with Claim-Based Settlement
 
 ```js
-{
-  "maxBalance": "highest balance allowed",
-  "prefix": "prefix for ilp address",
-  "secret": "your secret. your peer must know the corresponding public key.",
-  "peerPublicKey": "base64url encoded public key of your peer",
-  "rpcUri": "https://example.com/rpc", // this is used to communicate with your peer (see below)
-  "_store": { /* this object will be created by the connector */ },
-  "info": { /* the object to be returned by getInfo */ }
-}
-```
+const { MakePluginVirtual } = require('ilp-plugin-virtual')
+const { NotAcceptedError } = require('ilp-plugin-shared').Errors
+const Network = require('some-example-network')
+const BigNumber = require('bignumber.js')
 
-# Receiving RPC calls
+return MakePluginVirtual({
+  connect: async function (ctx, opts) {
+    await Network.connectToNetwork()
 
-In order to use plugin virtual, you need an HTTP endpoint that passes calls to the plugin.
-The url must take one query parameter (`method`), as well as a JSON body. The method parameter
-and parsed body must be passed to the plugin like so:
+    ctx.state.bestClaim = ctx.backend.getMaxValueTracker('incoming_claim')
+    ctx.state.maxInFlight = opts.maxInFlight
+  },
 
-```js
-plugin.receive(method, body)
-```
+  handleIncoming: async function (ctx, transfer) {
+    const incoming = await ctx.transferLog.getHighestIncomingBalance()
+    const bestClaim = await ctx.state.bestClaim.getMax() || { value: '0', data: null }
 
-# Generate a trustline prefix
+    const exceeds = new BigNumber(incoming)
+      .sub(bestClaim.value)
+      .gt(ctx.state.maxInFlight)
 
-If you want to generate a trustline prefix from a secret key, peer's public key, and currency,
-you can access the method through the token class:
+    if (exceeds) {
+      throw new NotAcceptedError(transfer.id + ' exceeds max in flight balance')
+    }
+  },
 
-```js
-const Token = require('ilp-plugin-virtual/src/util/token')
-const prefix = Token.prefix({
-  secretKey: 'o9Lt0oZFek1ArM_A0HUAq8M8edRuoSeLjX8i10sVUiY',
-  peerPublicKey: 'KRixgcBCBdyQln7IBYiopjuNO78QSFtXgOwP1sbsCSk',
-  currency: 'USD'
+  settleToBalance: async function (ctx, outgoingBalance) {
+    const claim = Network.createClaim(outgoingBalance)
+
+    return {
+      balance: outgoingBalance,
+      claim: claim
+    }
+  },
+
+  handleSettlement: async function (ctx, settlement) {
+    const { balance, claim } = settlement
+
+    if (Network.verify(claim, balance)) {
+      await ctx.state.bestClaim.setIfMax({ value: balance, data: claim })
+    }
```

Comment:
If the claim isn't valid, it should just be ignored. It's the responsibility of `handleIncomingPrepare` to make sure the best claim doesn't get too far from the incoming fulfilled and prepared.